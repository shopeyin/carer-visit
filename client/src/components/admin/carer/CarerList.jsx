import React from 'react';
import { Link } from 'react-router-dom';

function CarerList({ carers, handleDeleteCarer }) {
  let itemsToRender;
  if (carers && carers.length > 0) {
    itemsToRender = carers.map((carer) => {
      return (
        <div className="row" key={carer._id}>
          <div className="col-7 col-sm-5 col-md-3   m-1 carer__container-item">
            <Link to={`${carer._id}`} className="link-color">
              <div className="card" style={{ width: '100%' }}>
                <div className="card-body">
                  <h4 className="card-title text-center text-uppercase">
                    {' '}
                    {carer.name}
                  </h4>
                </div>
              </div>
            </Link>{' '}
          </div>
          <div className="col-3 col-sm-1 col-md-3 text-center p-4 trash-can">
            <i
              className="fa-solid fa-trash-can mt-2"
              onClick={() => {
                handleDeleteCarer(carer._id);
              }}
            ></i>
          </div>
        </div>
      );
    });
  } else if (carers && carers.length === 0) {
    itemsToRender = 'No carer';
  } else {
    itemsToRender = 'Loading...';
  }
  return (
    <div className="carer__container">
      <div className="row">
        <div className="col-11 col-sm-5 m-3 ">
          <Link to="add-carer" className="link-color navigation__link">
            Add Carer
          </Link>
        </div>
      </div>
      {itemsToRender}
    </div>
  );
}

export default CarerList;
