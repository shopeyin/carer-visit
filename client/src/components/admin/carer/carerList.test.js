import React from 'react';
import { shallow } from 'enzyme';
import CarerList from './CarerList';

describe('CarerList component', () => {
  describe('CarerList component', () => {
    let wrapper;
    let carers = [
      { _id: '1', name: 'Carer 1' },
      { _id: '2', name: 'Carer 2' },
    ];
    let handleDeleteCarerMock = jest.fn();

    beforeEach(() => {
      wrapper = shallow(
        <CarerList carers={carers} handleDeleteCarer={handleDeleteCarerMock} />
      );
    });

    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render carers when carers is not null', () => {
      expect(wrapper.find('.carer__container-item').length).toEqual(2);
    });

    it('should call handleDeleteCarer when trash icon is clicked', () => {
      wrapper.find('.fa-trash-can').first().simulate('click');
      expect(handleDeleteCarerMock).toHaveBeenCalledWith('1');
    });
  });

  it('should render  when carers is an empty array', () => {
    let carers = [];
    let handleDeleteCarerMock = jest.fn();
    let wrapper2 = shallow(
      <CarerList carers={carers} handleDeleteCarer={handleDeleteCarerMock} />
    );
    expect(wrapper2.find('.carer__container-item').length).toEqual(0);
  });
});
