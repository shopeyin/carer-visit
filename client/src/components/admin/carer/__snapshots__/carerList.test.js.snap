// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarerList component CarerList component should render correctly 1`] = `
<div
  className="carer__container"
>
  <div
    className="row"
  >
    <div
      className="col-11 col-sm-5 m-3 "
    >
      <Link
        className="link-color navigation__link"
        to="add-carer"
      >
        Add Carer
      </Link>
    </div>
  </div>
  <div
    className="row"
    key="1"
  >
    <div
      className="col-7 col-sm-5 col-md-3   m-1 carer__container-item"
    >
      <Link
        className="link-color"
        to="1"
      >
        <div
          className="card"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <div
            className="card-body"
          >
            <h4
              className="card-title text-center text-uppercase"
            >
               
              Carer 1
            </h4>
          </div>
        </div>
      </Link>
       
    </div>
    <div
      className="col-3 col-sm-1 col-md-3 text-center p-4 trash-can"
    >
      <i
        className="fa-solid fa-trash-can mt-2"
        onClick={[Function]}
      />
    </div>
  </div>
  <div
    className="row"
    key="2"
  >
    <div
      className="col-7 col-sm-5 col-md-3   m-1 carer__container-item"
    >
      <Link
        className="link-color"
        to="2"
      >
        <div
          className="card"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <div
            className="card-body"
          >
            <h4
              className="card-title text-center text-uppercase"
            >
               
              Carer 2
            </h4>
          </div>
        </div>
      </Link>
       
    </div>
    <div
      className="col-3 col-sm-1 col-md-3 text-center p-4 trash-can"
    >
      <i
        className="fa-solid fa-trash-can mt-2"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;
